src/matchbox/server/postgresql/utils/db.py

```diff
@@ -50,6 +57,44 @@ def get_model_subgraph(engine: Engine) -> rx.PyDiGraph:
     return G
 
 
+def get_data_subgraph(engine: Engine) -> rx.PyDiGraph:
+    """Retrieves the complete data subgraph as a PyDiGraph."""
+    G = rx.PyDiGraph()
+    nodes = {}
+
+    with Session(engine) as session:
+        sources = {source.model: source for source in session.query(Sources).all()}
+
+        for source in sources.values():
+            source_id = f"{source.schema}.{source.table}"
+            if source_id not in nodes:
+                source_idx = G.add_node({"id": source_id, "type": "source"})
+                nodes[source_id] = source_idx
+
+        for cluster in session.query(Clusters).all():
+            cluster_id = cluster.hash
+            if cluster_id not in nodes:
+                cluster_idx = G.add_node({"id": cluster_id, "type": "cluster"})
+                nodes[cluster_id] = cluster_idx
+
+            if cluster.id is not None and cluster.dataset is not None:
+                source = sources.get(cluster.dataset)
+                if source:
+                    data_id = str(cluster.id)
+                    data_idx = G.add_node({"id": data_id, "type": "data"})
```

Comment:
Just curious: why do you have a separate node for a data point and a cluster of 1?